core/controllers/contributor_dashboard.py

```diff
@@ -560,20 +691,25 @@ def get(self):
         self.render_json(self.values)
 
 
-class UserContributionRightsDataHandler(base.BaseHandler):
+class UserContributionRightsDataHandler(
+    base.BaseHandler[Dict[str, str], Dict[str, str]]
+):
     """Provides contribution rights of the logged in user in translation,
     voiceover and question category on the contributor dashboard.
     """
 
     GET_HANDLER_ERROR_RETURN_TYPE = feconf.HANDLER_TYPE_JSON
-    URL_PATH_ARGS_SCHEMAS = {}
-    HANDLER_ARGS_SCHEMAS = {'GET': {}}
+    URL_PATH_ARGS_SCHEMAS: Dict[str, str] = {}
+    HANDLER_ARGS_SCHEMAS: Dict[str, Dict[str, str]] = {'GET': {}}
 
     @acl_decorators.open_access
-    def get(self):
+    def get(self) -> None:
         """Handles GET requests."""
         contribution_rights = None
         if self.username:
+            # Here we are sure that 'user_id' is not None because if 'username'
+            # exists then 'user_id' is also going to exists.
```

Comment:
exists --> exist

Why don't we just do "if self.user_id is not None" instead of "if self.username"?

Reply:
Because `self.username` can only be recorded if a user is logged in correctly.
By this I mean,
if a user logged in with deleted account ( or is supposed to be deleted ) then its username not recoded https://github.com/oppia/oppia/blob/934e23416ebb513935c124ed2be93e26a2632785/core/controllers/base.py#L259
If a user is partially logged in then also username is not recorded
https://github.com/oppia/oppia/blob/934e23416ebb513935c124ed2be93e26a2632785/core/controllers/base.py#L261
Iff user is correctly logged in then username recorded:
https://github.com/oppia/oppia/blob/934e23416ebb513935c124ed2be93e26a2632785/core/controllers/base.py#L265

But in all cases `self.user_id` is recorded:
https://github.com/oppia/oppia/blob/934e23416ebb513935c124ed2be93e26a2632785/core/controllers/base.py#L257

Reply:
> (a)

Done

> (b)

Done